integrationtests/src/test/java/org/neo4j/causalclustering/scenarios/ListQueriesProcedureInClusterIT.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.scenarios;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.neo4j.causalclustering.core.CoreGraphDatabase;
+import org.neo4j.causalclustering.discovery.Cluster;
+import org.neo4j.causalclustering.readreplica.ReadReplicaGraphDatabase;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.impl.factory.GraphDatabaseFacade;
+import org.neo4j.test.causalclustering.ClusterRule;
+import org.neo4j.test.rule.VerboseTimeout;
+import org.neo4j.test.rule.concurrent.ThreadingRule;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
+
+public class ListQueriesProcedureInClusterIT {
+    private final ClusterRule clusterRule =
+            new ClusterRule(getClass()).withNumberOfCoreMembers(3).withNumberOfReadReplicas(1);
+    private final VerboseTimeout timeout = VerboseTimeout.builder()
+            .withTimeout(1000, TimeUnit.SECONDS)
+            .build();
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(clusterRule).around(timeout);
+
+    private Cluster cluster;
+
+    @Rule
+    public final ThreadingRule threads = new ThreadingRule();
+
+    @Before
+    public void setup() throws Exception {
+        cluster = clusterRule.startCluster();
+    }
+
+    @Test
+    public void shouldListQueriesOnCoreAndReplica() throws Exception {
```

Comment:
I really would like it if this test was broken down into smaller method with descriptive names. There is too much code to make it easy to get an overview of what is actually being tested.

Breaking the method up will also make it cleaner to break the flow up in a simple sequence of transactions, avoiding the trouble of overlapping transactions that emerge from nesting the code as it is written now.